src/modules/SdnDiag.Utilities/private/Invoke-WebRequestWithRetry.ps1

```diff
@@ -80,19 +80,12 @@ function Invoke-WebRequestWithRetry {
             break
         }
         catch {
-            if ($_.Exception.Response.StatusCode -eq "NotFound") {
-                "{0} ({1})" -f $_.Exception.Message, $_.Exception.Response.ResponseUri.AbsoluteUri | Trace-Output -Level:Warning
-                return $null
-            }
-            else {
-                $_ | Trace-Output -Level:Error
-            }
-
             if (($counter -le $MaxRetry) -and $Retry) {
                 "Retrying operation in {0} seconds. Retry count: {1}." - $RetryIntervalInSeconds, $counter | Trace-Output
                 Start-Sleep -Seconds $RetryIntervalInSeconds
             }
             else {
+                $_ | Trace-Exception
```

Comment:
Not related to this change thinking, whether we need two helper function `Invoke-RestMethodWithRetry` and `Invoke-WebRequestWithRetry`?

Reply:
I had evaluated if we needed both at one point, but decided to just keep both since I had written up the logic to work. There are also now 8 references to `Invoke-WebRequestWithRetry` so it would be a bit more of a PR that I would do outside of this one if we did consolidate.

Reply:
I had evaluated if we needed both at one point, but decided to just keep both since I had written up the logic to work. There are also now 8 references to `Invoke-WebRequestWithRetry` so would be a bit more of a PR that I would do outside of this one if we did consolidate.

![image](https://github.com/microsoft/SdnDiagnostics/assets/18577812/493a8e10-289f-4f77-9d10-51feea240e48)